addons/account_payment/security/ir_rules.xml

```diff
@@ -1,16 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo noupdate="1">
 
-    <!-- Transactions -->
-
-    <record id="payment_transaction_billing_rule" model="ir.rule">
-        <field name="name">Access every transaction</field>
-        <field name="model_id" ref="payment.model_payment_transaction"/>
-        <!-- Reset the domain defined by payment.transaction_user_rule -->
-        <field name="domain_force">[(1, '=', 1)]</field>
-        <field name="groups" eval="[(4, ref('account.group_account_invoice'))]"/>
-    </record>
```

Comment:
As Pierre pointed out, you must write an upgrade script to force remove `ir.rule` records are they are flagged as `noupdate`.

It's not needed for `ir.model.access`, though.